examples/nlp/language_modeling/conf/megatron_gpt_config.yaml

```diff
@@ -154,6 +154,7 @@ model:
   # Distributed checkpoint setup
   dist_ckpt_format: 'zarr' # Set to 'torch_dist' to use PyTorch distributed checkpoint format.
   dist_ckpt_load_on_device: True # whether to load checkpoint weights directly on GPU or to CPU
+  dist_ckpt_parallel_save: False # if true, each worker will write its own part of the dist checkpoint
```

Comment:
can you rename to `dist_ckpt_fully_parallel_save` please?

Reply:
I'm leaning towards proceeding with the current name, I think the "fully" part is not fully required :).

Please let me know if you feel there's a strong reason to include it. Thanks.

Reply:
Dist-ckpt save is already parallel across processes. The word "fully" emphasizes the fact that we are using all ranks to do the saving. This is analogous to the "fully" in FSDP.
Besides, we have this nomenclature [in Megatron-LM already](https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/training/arguments.py#L1255) so it's good for consistency

Reply:
Hi, thanks for the comment, let me add some more context on my perspective,

if we focus on the actual saving of data from memory to disk files, then the `dp_zero_gather_scatter` function saves all data sequentially, the data is distributed but the saving is not. As a result, I want to avoid adding the "fully" part because the actual save is not partially parallel with `dp_zero_gather_scatter` (the data collection is parallel, but not the save part).

From the user's perspective, it might be misleading to have the fully part, because somebody might assume that having the option turned off there's still some part of the file that's saved in parallel, but IIUC that's not the case.